File: Array/977. squares-of-a-sorted-array/977.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left, right, k = 0, len(nums)-1, len(nums)-1
        res = [float('inf')] * len(nums)
        while left <= right:
            if((nums[left]**2)<(nums[right]**2)):
                res[k] = nums[right]**2
                k -= 1
                right -= 1
            else:
                res[k] = nums[left]**2
                left += 1
                k -= 1

        return res

File: Array/977. squares-of-a-sorted-array/977.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0, right = nums.size()-1, k = nums.size()-1;
        vector<int> result(nums.size(), 0);
        while(left<=right){
            if ((nums[left]*nums[left])<(nums[right]*nums[right])){
                result[k] = nums[right]*nums[right];
                k--;
                right--;
            }
            else{
                result[k] = nums[left] * nums[left];
                k--;
                left++;
            }
        }
        return result;
    }
};
